src/Language/Parser.php

```diff
@@ -1307,6 +1310,42 @@ protected function lexTypeSystemExtension(): TypeExtensionNodeInterface
         throw $this->unexpected($token);
     }
 
+    /**
+     * SchemaExtension :
+     *  - extend schema Directives[Const]? { OperationTypeDefinition+ }
+     *  - extend schema Directives[Const]
+     *
+     * @return SchemaExtensionNode
+     * @throws SyntaxErrorException
+     */
+    protected function lexSchemaExtension(): SchemaExtensionNode
+    {
+        $start = $this->lexer->getToken();
+
+        $this->expectKeyword(KeywordEnum::EXTEND);
+        $this->expectKeyword(KeywordEnum::SCHEMA);
+
+        $directives = $this->lexDirectives(true);
+
+        $parseFunction = function (): OperationTypeDefinitionNode {
```

Comment:
Does static return type-hinting really work for anonymous functions?

Reply:
Yes they do. :)